NIP-31 Pinned Events (#44)
nostr-protocol/nips: 31.md

NIP-31
======

Pinned Events
-------------

`draft` `optional` `author:plantimals`

NIP-31 amends the content metadata for `Kind 0` as described in [NIP-01](01.md).

> Each pubkey MAY indicate a pinned event by setting the field `"pinned_event"`

Reviewer: This should use `e` tags instead with a `pinned` marker, otherwise only 1 pinned event is allowed. Clients can choose to display 1 or more.
e.g.
```
{
  ...
  "tags": [["e","<eventId>", "<relayUrl>", "pinned"]]
  ...
}
```

in the json content of one's `Kind 0` metadata event to point to an
event `ID`.

> Example json content from [this event](https://nostr.io/api/v1/events/71412d5cdb7e675b9806d23bda3d7c4878ca7f8e72ae3477c7179938e857cb9b):

Reviewer 2: The link doesn't work right now.

```suggestion
Example json content:
```

```json
{
    "name":"plantimals",

>     "picture":"https://plantimals.org/img/avatar.png","pinned_event":"c76166d9384355e8f55abfc071294fb815ac5f4050e2b9373b997b4bed66c669",

Reviewer 2: ```suggestion
    "picture":"https://plantimals.org/img/avatar.png",
    "pinned_event":"c76166d9384355e8f55abfc071294fb815ac5f4050e2b9373b997b4bed66c669",
```

    "about":"[plantimals](https://plantimals.org)",
    "nip05":"_@plantimals.org"
}
```

The `pinned_event` field is meant to provide a link to additional content for

> someone wishing to supplement their `about` text. Users SHOULD ONLY pin events
> that they have authored in order to avoid mistaken identity.

Reviewer 2: ```suggestion
someone wishing to supplement their `about` text. Clients SHOULD ONLY show pinned
events as such if they were authored by the same account as the `kind 0` event.
```

Reviewer 3: I agree, let's pin stuff from others, why not? Clients are still free to ignore these if that makes things harder for them.

Author: I'm not opposed to it. in fact, I added the language based on my incorrect reading of Vishal's comments here: https://github.com/nostr-protocol/nips/pull/42#issuecomment-1229578286
